pkg/agent/proxy/proxier.go

```diff
@@ -387,12 +392,26 @@ func (p *proxier) installServices() {
 			}
 			for _, ingress := range toAdd {
 				if ingress != "" {
-					if err := p.installLoadBalancerServiceFlows(groupID, net.ParseIP(ingress), uint16(svcInfo.Port()), svcInfo.OFProtocol, uint16(svcInfo.StickyMaxAgeSeconds())); err != nil {
+					if err := p.installLoadBalancerServiceFlows(groupID, net.ParseIP(ingress), uint16(svcInfo.Port()), svcInfo.OFProtocol, p.gwConfig, uint16(svcInfo.StickyMaxAgeSeconds())); err != nil {
 						klog.Errorf("Error when installing LoadBalancer Service flows: %v", err)
 						continue
 					}
 				}
 			}
+
+			// Update route on the host for Cluster IP Service.
+			if svcInfo.ClusterIP() != nil {
+				svcIPNet := net.IPNet{IP: svcInfo.ClusterIP(), Mask: net.CIDRMask(32, 32)}
+				if needRemoval {
+					if err := p.deleteClusterIPServiceRoutes(&svcIPNet); err != nil {
+						klog.Errorf("Error when deleting route '%s' for Cluster IP Service: %v", svcIPNet.String(), err)
+					}
+				} else {
+					if err := p.addClusterIPServiceRoutes(&svcIPNet, p.hostname, nil, p.gwConfig); err != nil {
+						klog.Errorf("Error when adding route '%s' for Cluster IP Service: %v", svcIPNet.String(), err)
```

Comment:
please switch to structured logging (`klog.ErrorS`) for new code

Reply:
Updated.